contracts/coordinator/src/state.rs

```diff
@@ -79,53 +79,51 @@ pub fn save_chain_contracts(
     Ok(())
 }
 
-#[allow(dead_code)] // Used in tests, might be useful in future query
 pub fn contracts_by_chain(
     storage: &dyn Storage,
-    chain_name: ChainName,
+    chain_name: &ChainName,
 ) -> Result<ChainContractsRecord, ContractError> {
     CHAIN_CONTRACTS_MAP
         .may_load(storage, chain_name.clone())?
-        .ok_or(ContractError::ChainNotRegistered(chain_name))
+        .ok_or(ContractError::ChainNotRegistered(chain_name.clone()))
 }
 
-#[allow(dead_code)] // Used in tests, might be useful in future query
 pub fn contracts_by_prover(
     storage: &dyn Storage,
-    prover_address: ProverAddress,
+    prover_address: &ProverAddress,
 ) -> Result<ChainContractsRecord, ContractError> {
     CHAIN_CONTRACTS_MAP
         .idx
         .by_prover
         .item(storage, prover_address.clone())?
         .map(|(_, record)| record)
-        .ok_or(ContractError::ProverNotRegistered(prover_address))
+        .ok_or(ContractError::ProverNotRegistered(prover_address.clone()))
 }
 
-#[allow(dead_code)] // Used in tests, might be useful in future query
 pub fn contracts_by_gateway(
     storage: &dyn Storage,
-    gateway_address: GatewayAddress,
+    gateway_address: &GatewayAddress,
 ) -> Result<ChainContractsRecord, ContractError> {
     CHAIN_CONTRACTS_MAP
         .idx
         .by_gateway
         .item(storage, gateway_address.clone())?
         .map(|(_, record)| record)
-        .ok_or(ContractError::GatewayNotRegistered(gateway_address))
+        .ok_or(ContractError::GatewayNotRegistered(gateway_address.clone()))
 }
 
-#[allow(dead_code)] // Used in tests, might be useful in future query
 pub fn contracts_by_verifier(
     storage: &dyn Storage,
-    verifier_address: VerifierAddress,
+    verifier_address: &VerifierAddress,
 ) -> Result<ChainContractsRecord, ContractError> {
     CHAIN_CONTRACTS_MAP
         .idx
         .by_verifier
         .item(storage, verifier_address.clone())?
         .map(|(_, record)| record)
-        .ok_or(ContractError::VerifierNotRegistered(verifier_address))
+        .ok_or(ContractError::VerifierNotRegistered(
+            verifier_address.clone(),
+        ))
 }
```

Comment:
in most cases it's best to propagate ownership concerns up the chain. In particular, try not to make decisions about cloning for the caller. They might not need the value after the call to these functions anymore and could just relent ownership so the data wouldn't need to be cloned